Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -50,7 +50,7 @@ class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor<Site>, SearchViewController
             let deferred = self.profile.history.getSitesByFrecencyWithHistoryLimit(100, bookmarksLimit: 5, whereURLContains: query)
             inProgress = deferred as? Cancellable
 
-            deferred.uponQueue(DispatchQueue.main) { result in
+            deferred.upon() { result in
                 self.inProgress = nil
```

Comment:
Is it possible for us to drop a value of `inProgress` on the floor now that these are running asynchronously? After all, we're mutating `inProgress` from two threads.

Reply:
Good catch on the shared mutable `inProgress`, shouldn't this line be changed to:
```
DispatchQueue.main.async {
 self.inProgress = nil
}
```